File: src/getRandomInt.ts
type GetRandomIntOpts = {
  min?: number
  max: number
}
export const getRandomInt = (opts: GetRandomIntOpts) => {
  const { min = 0, max } = opts

  const byteSize = 16

  const range = max - min

  const a = new Uint16Array(1)
  crypto.getRandomValues(a)

  const randomNorm = a[0]! / (Math.pow(2, byteSize) + 1)

  // Inclusive
  const randomInter = Math.round(randomNorm * range + min)

  return randomInter
}

File: cpp/src/util/getRandomInt.cpp
#include "getRandomInt.h"
#include <random>

int getRandomInt(GetRandomIntOpts opts) {
  static std::random_device randomDevice;
  static std::mt19937 mtGen(randomDevice());

  std::uniform_int_distribution<int> uniformDist(opts.min, opts.max);
  auto number = uniformDist(mtGen);
  return number;
}
